src/index.jst

```diff
@@ -20,25 +24,19 @@ module.exports = function equal(a, b) {
 {{? it.es6 }}
     if (a instanceof Map) {
       if (a.size !== b.size) return false;
+      for(i of a.entries())
+        if (!b.has(i[0])) return false;
 
-      keys = getKeys(a);
-      for (i = a.size; i-- !== 0;)
-        if (!b.has(keys[i])) return false;
-
-      for (i = a.size; i-- !== 0;) {
-        key = keys[i];
-        if (!equal(a.get(key), b.get(key))) return false;
-      }
+      for(i of a.entries())
+        if (!equal(i[1], b.get(i[0]))) return false;
```

Comment:
Optimisation here made by reducing iterations amount, from three to two.
Plus, due to inlined `for` cycle - omitted `a.get(key)`, because item value is already available due to iteration process.